modules/sfp_c99.py

```diff
@@ -0,0 +1,624 @@
+
+            if latitude is not None and longitude is not None:
+                evt = SpiderFootEvent(
+                    "PHYSICAL_COORDINATES",
+                    f"{latitude}, {longitude}",
+                    self.__name__,
+                    event,
+                )
+                self.notifyListeners(evt)
+
+            if (
+                continent is not None
+                or country is not None
+                or region is not None
+                or city is not None
+                or postalCode is not None
+            ):
+                evt = SpiderFootEvent(
+                    "GEOINFO",
+                    f"Continent: {continent}, Country: {country}, Region: {region}, City: {city}, Postal code: {postalCode}",
+                    self.__name__,
+                    event,
+                )
+                self.notifyListeners(evt)
+
+    def emitSkypeResolverData(self, data, event):
+        self.emitRawRirData(data, event)
+
+        IP = data.get("ip")
+        IPs = data.get("ips")
+
+        if IP is not None and IP not in IPs:
+            evt = SpiderFootEvent(
+                "IP_ADDRESS",
+                IP,
+                self.__name__,
+                event,
+            )
+            self.notifyListeners(evt)
+
+        if isinstance(IPs, list):
+            for IPElem in IPs:
+                if self.checkForStop():
+                    return None
+
+                evt = SpiderFootEvent(
+                    "IP_ADDRESS",
+                    IPElem.strip(),
+                    self.__name__,
+                    event,
+                )
+                self.notifyListeners(evt)
+
+    def emitWafDetectorData(self, data, event):
+        self.emitRawRirData(data, event)
+        firewall = data.get("result")
+
+        if firewall is not None:
+            evt = SpiderFootEvent(
+                "WEBSERVER_TECHNOLOGY",
+                f"Firewall detected: {firewall}",
+                self.__name__,
+                event,
+            )
+            self.notifyListeners(evt)
+
+    def handleEvent(self, event):
+        eventName = event.eventType
+        srcModuleName = event.module
+        eventData = event.data
+
+        # Once we are in this state, return immediately.
+        if self.errorState:
+            return None
+
+        self.sf.debug(f"Received event, {eventName}, from {srcModuleName}")
+
+        if self.opts["api_key"] == "":
+            self.sf.error("You enabled sfp_c99, but did not set an API key!", False)
+            self.errorState = True
+            return None
+
+        if (
+            self.opts["response_type"].lower() != "plain"
+            and self.opts["response_type"].lower() != "json"
+        ):
+            self.sf.info(
+                "You enabled sfp_c99 ,but did not set response type to json or plain! JSON will be used by default"
+            )
+            self.opts["response_type"] = "json"
```

Comment:
As per my earlier comment, I suggest we stick with JSON and drop all the logic associated with `response_type`.